# the connecting root frame will be named like this
connecting_frame_id:
  "common_origin"

# these are the tf subtrees to be connected through the respective root frames so that the equal frames are equal
connections:
  - root_frame_id: "uav1/fcu"
    equal_frame_id: "uav1/gps_origin"

  - root_frame_id: "uav2/fcu"
    equal_frame_id: "uav2/gps_origin"
    # these offsets will be applied to the transformation from the connecting frame to each of the root frames
    offsets:
      # these are applied in the root frame (typically fcu frame, hence intrinsic)
      intrinsic:
        [
          [0.0, 0.0, 0.0, 0.0], # four numbers are interpreted as x,y,z,yaw
          [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0], # seven numbers are interpreted as x,y,z,qx,qy,qz,qw (the last four are a quaternion)
        ]
      # these are applied in the leaf (equal) frame (typically static frame, hence extrinsic)
      extrinsic:
        [
          [0.0, 0.0, 0.0, 0.0], # four numbers are interpreted as x,y,z,yaw
          [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0], # seven numbers are interpreted as x,y,z,qx,qy,qz,qw (the last four are a quaternion)
        ]

# if true, older messages than the last update of the respective frame connection will be ignored
ignore_older_messages: false

# if no update of a certain connection happens for this duration, it will be forced
max_update_period: 0.1 # seconds (set to zero to disable)
